x-pack/metricbeat/module/aws/awshealth/awshealth.go

```diff
@@ -0,0 +1,314 @@
+		if config.AWSConfig.FIPSEnabled {
+			o.EndpointOptions.UseFIPSEndpoint = awssdk.FIPSEndpointStateEnabled
+		}
+	})
+	events := m.getEventDetails(ctx, health_client)
+	for _, event := range events {
+		report.Event(event)
+	}
+
+	return nil
+}
+
+// getEventDetails retrieves AWS health events and their details using the provided AWS Health client.
+// It returns a list of Metricbeat events containing relevant AWS health information.
+func (m *MetricSet) getEventDetails(
+	ctx context.Context,
+	awsHealth *health.Client,
+) []mb.Event {
+	// Define event filter to fetch only upcoming and open events
+	eventFilter := types.EventFilter{
+		EventStatusCodes: []types.EventStatusCode{
+			types.EventStatusCodeUpcoming,
+			types.EventStatusCodeOpen,
+		},
+	}
+	var (
+		deEvents          []types.Event
+		affPage           health.DescribeAffectedEntitiesPaginator
+		healthDetails     []AWSHealthMetric
+		healthDetailsTemp []AWSHealthMetric
+		affEntityTemp     AffectedEntityDetails
+		affInputParams    health.DescribeAffectedEntitiesInput
+	)
+
+	// Create an instance of DescribeEventsInput with desired parameters
+	deInputParams := health.DescribeEventsInput{
+		Filter: &eventFilter,
+	}
+
+	// Define options for DescribeEventsPaginator
+	deOptions := &health.DescribeEventAggregatesPaginatorOptions{
+		Limit:                10,
+		StopOnDuplicateToken: true,
+	}
+
+	// Function option to apply options to the paginator
+	deOptFn := func(options *health.DescribeEventsPaginatorOptions) {
+		// Apply the provided options
+		options.Limit = deOptions.Limit
+		options.StopOnDuplicateToken = deOptions.StopOnDuplicateToken
+	}
+	// Define options for DescribeAffectedEntitiesPaginator
+	affOptions := &health.DescribeAffectedEntitiesPaginatorOptions{
+		Limit:                10,
+		StopOnDuplicateToken: true,
+	}
+	// Function option to apply options to the paginator
+	affOptFn := func(options *health.DescribeAffectedEntitiesPaginatorOptions) {
+		// Apply the provided options
+		options.Limit = affOptions.Limit
+		options.StopOnDuplicateToken = affOptions.StopOnDuplicateToken
+	}
+	// Create DescribeEventsPaginator with AWS Health client and options
+	dePage := health.NewDescribeEventsPaginator(awsHealth, &deInputParams, deOptFn)
+
+	for dePage.HasMorePages() {
+		healthDetailsTemp = []AWSHealthMetric{}
+
+		// Perform actions for the current page
+		currentPage, err := dePage.NextPage(ctx)
+		if err != nil {
+			m.Logger().Errorf("[AWS Health] DescribeEvents failed with : %w", err)
+			break
+		}
+		deEvents = currentPage.Events
+		eventArns := make([]string, len(deEvents))
+		// Iterate through events to extract relevant information
+		for i := range deEvents {
+			healthDetailsTemp = append(healthDetailsTemp, AWSHealthMetric{
+				EventArn:          awssdk.ToString(deEvents[i].Arn),
+				EndTime:           awssdk.ToTime(deEvents[i].EndTime),
+				EventScopeCode:    awssdk.ToString((*string)(&deEvents[i].EventScopeCode)),
```

Comment:
this can be simplified to `string(deEvents[i].EventScopeCode)`

Reply:
(same thing in multiple other places)

Reply:
Made changes as per the suggestion.

updated code example

```
AwsAccountId:    awssdk.ToString(ace.AwsAccountId),
EntityUrl:       awssdk.ToString(ace.EntityUrl),
EntityValue:     awssdk.ToString(ace.EntityValue),
```